.github/workflows/build-and-test.yml

```diff
@@ -38,18 +38,10 @@ jobs:
 
   dispatch-trivy:
     needs: [ build ]
-    runs-on: ubuntu-latest
-
+    uses: ./.github/workflows/trivy.yml
+    secrets: inherit
```

Comment:
I was reading https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow and there are ways to explicitly declare which secrets a task needs and also explicitly pass in secrets. Would that work here & make sense to do?